examples/express/src/index.ts

```diff
@@ -31,46 +29,58 @@ function resolveAfterHalfSecond(): Promise<string> {
   });
 }
 
-async function asyncCall() {
+const asyncCall = autometrics(async function asyncCall() {
   console.log("Calling async function");
   return await resolveAfterHalfSecond();
-}
+});
 
 app.get("/", autometrics(rootRoute));
-app.get("/bad", (req, res) => badRouteMetrics(req, res));
+app.get("/bad", autometrics(badRoute));
 app.get("/async", autometrics(asyncRoute));
 
 app.listen(port, () => console.log(`Example app listening on port ${port}!`));
 
 async function generateRandomTraffic() {
-  const loopTimes = Math.floor(Math.random() * 30 + 20);
   const http = await import("http");
+  console.log("Generating random traffic");
 
-  for (let i = 0; i < loopTimes; i++) {
+  while (true) {
     const type = Math.floor(Math.random() * 3);
 
+    // sleep duration is between 10 and 50 ms
+    const sleepDuration = Math.floor(Math.random() * 400 + 100);
+
+    console.log(`Sleeping for ${sleepDuration}ms`);
+
     switch (type) {
       case 0: {
-        http.get("http://localhost:8080");
-        break;
+        console.log("GET /");
+        await fetch("http://localhost:8080");
       }
 
       case 1: {
-        http.get("http://localhost:8080/async");
-        break;
+        console.log("GET /async");
+        await fetch("http://localhost:8080/async");
       }
 
       case 2: {
-        http.get("http://localhost:8080/bad");
-        break;
+        console.log("GET /bad");
+        await fetch("http://localhost:8080/bad");
       }
 
-      default:
-        break;
+      default: {
+        console.log("none");
+      }
     }
+
+    sleep(sleepDuration);
   }
 }
 
+const sleep = (ms: number) => {
```

Comment:
In the async world, this is more commonly called `delay()`, since it doesn’t cause the thread or process to sleep.